api/sip/client.py

```diff
@@ -705,14 +707,22 @@ def read_message(self, max_size=1024*1024):
         data = ""
         tmp = ""
         while not done:
-            tmp = self.socket.recv(4096)
+            try:
+                tmp = self.socket.recv(4096)
+            except socket.timeout:
+                raise socket.error
+            except socket.error, e:
+                raise IOError("Socket error occurred receiving SIP response: %s" % e)
+            except Exception, e:
+                raise IOError("Some other error occurred with SIP response: %s" % e.message)
```

Comment:
So I feel like the only new exception handling code is here -- the code that takes any kind of exception, even if it's not an IOError, and turns it into an IOError. And that doesn't seem like it jibes with your problem.

I suspect your problem was the socket had _no_ timeout and that was causing a different type of error which wasn't caught here? I'm testing that locally now but I don't really get what happens when the socket errors are reraised as `IOError`.

Reply:
Looking at the code, I think the problem was that the lack of a timeout was causing the process to be killed rather than an exception to be raised. If a timeout is in place, the existing code should handle the resulting exception.

Reply:
Yes, agreed about the no timeout. My thinking is that the original problem had to do with a timeout on the remote end, where the socket was closed, but somehow not detected on our end. One would think that would have triggered the except (IOError, socket.error), but it didn't. Originally, without a timeout specified for the socket, the recv was blocking, and only the UWSGI timeout killed it by killing the worker. So the blocks I added were mainly designed to help me see what happened when I added a timeout. Without a thorough knowledge of socket programming/behavior, I need comments like these you've made to see where to go with this. Thanks.